e2e/image/pull_test.go

```diff
@@ -0,0 +1,75 @@
+package image
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/stretchr/testify/require"
+)
+
+const notaryURL = "https://notary-server:4443"
+const registryPrefix = "registry:5000"
+
+const alpineImage = "registry:5000/alpine:3.6"
+const busyboxImage = "registry:5000/busybox:1.27.2"
+
+func TestPullWithContentTrust(t *testing.T) {
+	image := createTrustedRemoteImage(t, "trust", "latest")
+
+	// test that pulling without the tag defaults to latest
+	imageWithoutTag := strings.TrimSuffix(image, ":latest")
+	icmd.RunCmd(trustedCmdNoPassphrases(icmd.Command("docker", "pull", imageWithoutTag))).Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "rmi", image).Assert(t, icmd.Success)
+
+	// try pulling with the tag, record output for comparison later
+	result := icmd.RunCmd(trustedCmdNoPassphrases(icmd.Command("docker", "pull", image)))
+	result.Assert(t, icmd.Success)
+	firstPullOutput := result.String()
+	icmd.RunCommand("docker", "rmi", image).Assert(t, icmd.Success)
+
+	// push an unsigned image on the same reference name, but with different content (busybox)
+	icmd.RunCommand("docker", "pull", busyboxImage).Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "tag", busyboxImage, image).Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "push", image).Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "rmi", image).Assert(t, icmd.Success)
```

Comment:
This tag/push/rmi does not seem relevant to the test case. Can we remove it?

Pushing a trusted image can be a separate case.

Reply:
we need the tag/push/rmi here to test that when you push a new image with the same name that is not signed, DCT will still use the signed version

Reply:
I've pushed a new commit to factor out this setup logic into a helper, as per our offline conversation.  Please let me know if there are outstanding issues to fix